src/rdkafka_cgrp.c

```diff
@@ -5151,6 +5177,18 @@ rd_kafka_cgrp_subscription_set(rd_kafka_cgrp_t *rkcg,
         }
 }
 
+
+static void rd_kafka_cgrp_subscription_unset(rd_kafka_cgrp_t *rkcg) {
```

Comment:
We can unify this function with `rd_kafka_cgrp_subscription_set` which should do all the necessary changes when changing a subscription, that are:

- destroying previous `rkcg_subscription` and `rkcg_subscription_regex`
- unsetting the flags RD_KAFKA_CGRP_F_SUBSCRIPTION and RD_KAFKA_CGRP_F_WILDCARD_SUBSCRIPTION
- for the group protocol `consumer` setting the flags RD_KAFKA_CGRP_CONSUMER_F_SUBSCRIBED_ONCE and RD_KAFKA_CGRP_CONSUMER_F_SEND_NEW_SUBSCRIPTION
- for the group protocol `consumer` if it's a wildcard subscription combine the regexes and remove them from the subscription
- we can remove `rd_kafka_metadata_cache_hint_rktparlist` as it's fixed in a different way with the subscription version and creates some problems. I'll change the corresponding PR for that.